gm4_better_crossbows/data/better_crossbows/functions/shoot/tropical.mcfunction

```diff
@@ -0,0 +1,14 @@
+# @s = player using the crossbow
+
+#summon tropical fish at player's location and copy variant
+summon minecraft:tropical_fish ~ ~1.2 ~ {FromBucket:1,Tags:["gm4_cb_proj"],ActiveEffects:[{Id:10,Amplifier:10,Duration:200,ShowParticles:0b}]}
+data modify entity @e[type=minecraft:tropical_fish,tag=gm4_cb_proj,distance=..2,limit=1] Variant set from entity @s Inventory[{Slot:-106b}].tag.BucketVariantTag
+
+#replace bucket of tropical fish in player's offhand with water bucket
+execute unless entity @s[gamemode=creative] run replaceitem entity @s[nbt={Inventory:[{Slot:-106b,id:"minecraft:tropical_fish_bucket"}]}] weapon.offhand minecraft:water_bucket

```

Comment:
```suggestion
execute unless entity @s[gamemode=creative] run replaceitem entity @s weapon.offhand minecraft:water_bucket
```
This nbt check isn't required as the item was already checked in `check_projectile.mcfunction`